src/main/java/com/transferwise/openbanking/client/api/payment/v3/AsyncPaymentClient.java

```diff
@@ -17,32 +17,37 @@
 import com.transferwise.openbanking.client.oauth.OAuthClient;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpEntity;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.client.RestClientException;
-import org.springframework.web.client.RestClientResponseException;
-import org.springframework.web.client.RestOperations;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientException;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+import wiremock.org.apache.commons.lang3.Validate;
 
 @Slf4j
 @SuppressWarnings("checkstyle:parametername")
-public class RestPaymentClient extends BasePaymentClient implements PaymentClient {
+public class AsyncPaymentClient extends BasePaymentClient implements PaymentClient {
 
     private static final String ENDPOINT_PATH_FORMAT = "%s/open-banking/v3.%s/pisp/%s";
 
     private static final String PAYMENT_CONSENT_RESOURCE = "domestic-payment-consents";
     private static final String PAYMENT_RESOURCE = "domestic-payments";
+    private static final String ON_ERROR_CREATE_PAYMENT_LOG = "Call to create payment consent endpoint failed";
+    public static final String ON_ERROR_SUBMIT_PAYMENT_LOG = "Call to submit payment endpoint failed";
+    public static final String ON_ERROR_GET_PAYMENT_CONSENT_LOG = "Call to get payment consent endpoint failed";
+    public static final String ON_ERROR_GET_PAYMENT_LOG = "Call to get payment endpoint failed";
+    public static final String ON_ERROR_GET_COF_LOG = "Call to get confirmation of funds endpoint failed";
```

Comment:
We can move this to a separate file saying ErrorLog.

Reply:
Done!